win32ss/user/user32/windows/cursoricon.c

```diff
@@ -2038,19 +2038,63 @@ HANDLE WINAPI CopyImage(
         return NULL;
     }
 
+    /* If hImage is NULL, then do fast return after setting LastError */
+    if (!hImage)
+    {
+        switch (uType)
+        {
+            case IMAGE_BITMAP:
+                SetLastError(ERROR_INVALID_HANDLE);
+                break;
+            case IMAGE_CURSOR:
+            case IMAGE_ICON:
+                SetLastError(ERROR_INVALID_CURSOR_HANDLE);
+                break;
+            default:
+                SetLastError(ERROR_INVALID_PARAMETER);
+        }
+        return NULL;
+    }
```

Comment:
Since now these two parts of code are successive, I would suggest you just incorporate inside the existing switch below these null-tests, see the following suggestions.
The default case that returns `ERROR_INVALID_PARAMETER` is already handled in the other switch.